apps/communications/contacts/js/activities.js

```diff
@@ -216,9 +217,46 @@ var ActivityHandler = {
     }
   },
 
+  // Generates a name for the contact returned as a vcard
+  _getContactName: function(theContact) {
+    var out = '';
```

Comment:
`out` can be an array then we push components and do a later `join` to avoid checking if  `out` has content or not later.